client/src/components/AddCohort/index.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import './style.css';
+import { Form, Input, Button, message } from 'antd';
+import postData from '../postData';
+
+const Demo = () => {
+  const onFinish = async (values) => {
+    const response = await postData('/api/v1/cohorts', values);
+    const { message: resMessage } = response.data.data;
+    if (response.status === 201) {
+      message.success(resMessage);
+    } else {
+      message.error(resMessage);
+    }
+  };
+
+  const onFinishFailed = () => {
```

Comment:
It would be better to make this a class component and then have `onFinish` and `onFinishFailed` be class methods (so you would use them as `this.onFinish`, `this.onFinishFailed` instead.
Otherwise, the way the code currently is, you are recreating an onFinish and onFinishFailed function in memory every time this component re-renders.

Reply:
@tomduggan85 could you explain more about this point, please?

Reply:
@MohammedAlghazali Sure (and sorry for the delay).
This whole function will run every time the AddCohort component is rendered - so every time its state or props change.
That means that the two functions, onFinishFailed and onFinish, will be redeclared every time AddCohort renders (every time props or state change).  This isn't that bad - the code still works, it doesn't cause bugs - but it's a little inefficient, because you're creating new copies of the exact same function in memory.  If you had a component that re-renders very often, it could be a real performance problem as more and more memory is taken up by these duplicate functions.
The way to solve this is by converting AddCohort to be a class component (`class AddCohort extends React.component {`, and then have onFinish and onFinishFailed be class methods on that class, because then only one instance of that function is created in memory for that component, and it's reused instead of recreated every time the component re-renders.

Reply:
@tomduggan85 I would like to thank you for clarifying this point
I think I am now more knowledgeable about the performance of react
And now I can dive deeper into this part.
thank you  ❤️

Reply:
@MohammedAlghazali afwan!